README.md

```diff
@@ -1,2 +1,32 @@
-# terraform-aws-iam-role-for-github-repository
-Terraform module creating an IAM role which can be assumed via OpenID by a GitHub-repository-specific GitHub Actions workflow
+# AWS IAM Role Assumable via OpenID by GitHub Actions Workflows
+
+Terraform module creating an IAM role which can be assumed via OpenID by a GitHub-repository-specific GitHub Actions workflow.
+
+https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services
+
+## Usage
+
+```tf
+module "iam_role" {
+  source  = "babbel/iam-role-for-github-repository/aws"
+  version = "~> 1.0"
+
+  github_repository           = github_repository.example
+  iam_openid_connect_provider = aws_iam_openid_connect_provider.github
+}
+
+resource "github_repository" "example" {
+  name = "example"
+}
+
+resource "aws_iam_openid_connect_provider" "github" {
+  url = data.tls_certificate.github.url
+
+  client_id_list  = ["sts.amazonaws.com"]
+  thumbprint_list = [data.tls_certificate.github.certificates.0.sha1_fingerprint]
```

Comment:
Isn't this going to cause the same problems we've seen elsewhere, unless we filter the list?

Reply:
right, I need to update the example.